### ISSUE TITLE
Error: Unexpected path returned for .net core

### ISSUE THREAD

**Issue by SergeiY** [microsoft/vscode-azurearmtools#396]
Repro steps:

1.
2.

Action: acquireDotnet
Error type: Error
Error Message: Unexpected path returned for .net core: c:\Users\Sergei\AppData\Roaming\Code\User\globalStorage\msazurermtools.azurerm-vscode-tools\.dotnet\2.2.5\dotnet.exe

Version: 0.8.2
OS: win32
Product: Visual Studio Code
Product Version: 1.38.1
Language: en

<details>
<summary>Call Stack</summary>

```
extension.bundle.js:1:505915extension.bundle.js:1:505915
o extension.bundle.js:1:500914
```

</details>

**Comment by StephenWeatherford** [comment]
Thanks for reporting, I've been trying to figure out what is causing this.  Could you do me a favor and let me know what files are in the following folders?

1. c:\Users\Sergei\AppData\Roaming\Code\User\globalStorage\msazurermtools.azurerm-vscode-tools.dotnet\
1. c:\Users\Sergei\AppData\Roaming\Code\User\globalStorage\msazurermtools.azurerm-vscode-tools.dotnet\2.2.5\

Thanks!

**Comment by SergeiY** [comment]
Hi, Thanks for quick feedback.C:\Users\Sergei\AppData\Roaming\Code\User\globalStorage\msazurermtools.azurerm-vscode-tools\.dotnet\2.2.5

There are no files inside the folder above.
Thanks
    On Thursday, 14 November 2019, 22:28:17 GMT, Stephen Weatherford (MSFT) <notifications@github.com> wrote:  
 
 
Thanks for reporting, I've been trying to figure out what is causing this. Could you do me a favor and let me know what files are in the following folders?
   
   - c:\Users\Sergei\AppData\Roaming\Code\User\globalStorage\msazurermtools.azurerm-vscode-tools.dotnet\
   - c:\Users\Sergei\AppData\Roaming\Code\User\globalStorage\msazurermtools.azurerm-vscode-tools.dotnet\2.2.5\

Thanks!

**Comment by StephenWeatherford** [comment]
Could you try installing dotnet core manually using this command line (replacing USERNAME twice)?  Thanks!
```
"c:\Users<USERNAME>\Repos\vscode-azurearmtools\assets\install scripts\dotnet-install.cmd" -InstallDir 'c:\Users<USERNAME>\AppData\Roaming\Code\User\globalStorage\msazurermtools.azurerm-vscode-tools.dotnet\2.2.5' -Runtime dotnet -Version 2.2.5
```

**Comment by StephenWeatherford** [comment]
@SergeiY Hey, sorry to bother you, wondering if you've had a chance to try the manual install.  Thanks!

**Comment by cschertz** [comment]
Having the same issue after running the manual install above this are the files in the directory 
c:\Users\<user>\AppData\Roaming\Code\User\globalStorage\msazurermtools.azurerm-vscode-tools\.dotnet>dir /s
 Volume in drive C is OSDisk
 Volume Serial Number is 3012-F2EC

 Directory of c:\Users\<user>\AppData\Roaming\Code\User\globalStorage\msazurermtools.azurerm-vscode-tools\.dotnet

10/25/2019  09:33    <DIR>          .
10/25/2019  09:33    <DIR>          ..
10/25/2019  09:33    <DIR>          2.2.5
               0 File(s)              0 bytes

 Directory of c:\Users\<user>\AppData\Roaming\Code\User\globalStorage\msazurermtools.azurerm-vscode-tools\.dotnet\2.2.5

10/25/2019  09:33    <DIR>          .
10/25/2019  09:33    <DIR>          ..
10/25/2019  09:33    <DIR>          shared
               0 File(s)              0 bytes

 Directory of c:\Users\<user>\AppData\Roaming\Code\User\globalStorage\msazurermtools.azurerm-vscode-tools\.dotnet\2.2.5\shared

10/25/2019  09:33    <DIR>          .
10/25/2019  09:33    <DIR>          ..
10/25/2019  09:33    <DIR>          Microsoft.NETCore.App
               0 File(s)              0 bytes

 Directory of c:\Users\<user>\AppData\Roaming\Code\User\globalStorage\msazurermtools.azurerm-vscode-tools\.dotnet\2.2.5\shared\Microsoft.NETCore.App

10/25/2019  09:33    <DIR>          .
10/25/2019  09:33    <DIR>          ..
10/25/2019  09:33    <DIR>          2.2.5
               0 File(s)              0 bytes

 Directory of c:\Users\<user>\AppData\Roaming\Code\User\globalStorage\msazurermtools.azurerm-vscode-tools\.dotnet\2.2.5\shared\Microsoft.NETCore.App\2.2.5

10/25/2019  09:33    <DIR>          .
10/25/2019  09:33    <DIR>          ..
10/25/2019  09:33            53,054 Microsoft.NETCore.App.deps.json
               1 File(s)         53,054 bytes

     Total Files Listed:
               1 File(s)         53,054 bytes
              14 Dir(s)  841,977,122,816 bytes free

**Comment by StephenWeatherford** [comment]
Ugh, major apologies, github's markdown interprets \. as an escaped period instead of backslash + period, so the backslash has been getting dropped in much of the conversation above.  Pretty sure here's what the installation command should look like (I'm on a Mac at the moment so can't test):
```
"c:\Users<USERNAME>\Repos\vscode-azurearmtools\assets\install scripts\dotnet-install.cmd" -InstallDir 'c:\Users<USERNAME>\AppData\Roaming\Code\User\globalStorage\msazurermtools.azurerm-vscode-tools\.dotnet\2.2.5' -Runtime dotnet -Version 2.2.5
```

Could you post the output of the installation, and the contents of `c:\Users<USERNAME>\AppData\Roaming\Code\User\globalStorage\msazurermtools.azurerm-vscode-tools\.dotnet\2.2.5` afterwards?  I'm really sorry for all the inconvenience.

If that's still not working, there's now a work-around available: Install dotnet core 2.2.x runtime (don't need the sdk) from here: https://dotnet.microsoft.com/download/dotnet-core/2.2, then set the `azureResourceManagerTools.languageServer.dotnetExePath` vscode setting to point to the installed dotnet.exe.  Please let us know how this goes.

Stephen

**Comment by StephenWeatherford** [comment]
This is now tracked by https://github.com/dotnet/vscode-dotnet-runtime/issues/50.

@cschertz @SergeiY @jdhuntington Are you still seeing this?  Thanks!
